From 5b0f52300d59fd9ca2df64dfb3bf8c194ca9dfef Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Sun, 07 Apr 2019 14:55:33 +0300
Subject: [PATCH] [SW-0] preboot: create get_cpu_boot_status_reg()

Since registers names are changed between devices:
Goya: mmPSOC_GLOBAL_CONF_WARM_REBOOT
Gaudi: mmPSOC_GLOBAL_CONF_CPU_BOOT_STATUS

Note that registers address is kept.
Change-Id: I85a5f052b35b93cc5f0eab81604931870989641f
---

diff --git a/common/preboot.c b/common/preboot.c
index cd58188..be7fce6 100644
--- a/common/preboot.c
+++ b/common/preboot.c
@@ -12,6 +12,7 @@
 #include "auto_generated.h"
 #include <stddef.h>
 #include <stdbool.h>
+#include "preboot.h"
 #include "uart.h"
 #include "spi.h"
 
@@ -141,7 +142,7 @@
 {
 	void (*spl_start)(void) = (void *)CONFIG_SPL_TEXT_BASE;
 
-	writel(CPU_BOOT_STATUS_IN_PREBOOT, mmPSOC_GLOBAL_CONF_WARM_REBOOT);
+	writel(CPU_BOOT_STATUS_IN_PREBOOT, get_cpu_boot_status_reg());
 
 	preboot_version();
 	preboot_pcie_init();
diff --git a/common/preboot.h b/common/preboot.h
new file mode 100644
index 0000000..9df3002
--- /dev/null
+++ b/common/preboot.h
@@ -0,0 +1,13 @@
+/*
+ * Copyright (C) 2017 HabanaLabs Ltd.
+ *   Author: Noam Camus <noamikong@gmail.com>
+ *
+ * SPDX-License-Identifier:     GPL-2.0+
+ */
+
+#ifndef COMMON_PREBOOT_H_
+#define COMMON_PREBOOT_H_
+
+unsigned int *get_cpu_boot_status_reg(void);
+
+#endif /* COMMON_PREBOOT_H_ */
diff --git a/gaudi/gaudi.c b/gaudi/gaudi.c
index c05e109..ae55ba5 100644
--- a/gaudi/gaudi.c
+++ b/gaudi/gaudi.c
@@ -26,3 +26,8 @@
 	return (unsigned int *)mmPSOC_GLOBAL_CONF_PAD_SEL_53;
 }
 
+#define mmPSOC_GLOBAL_CONF_CPU_BOOT_STATUS	(HL_CFG_BASE + 0xC4B300)
+unsigned int *get_cpu_boot_status_reg(void)
+{
+	return (unsigned int *)mmPSOC_GLOBAL_CONF_CPU_BOOT_STATUS;
+}
diff --git a/goya/goya.c b/goya/goya.c
index d5e5fe6..e423312 100644
--- a/goya/goya.c
+++ b/goya/goya.c
@@ -154,3 +154,7 @@
 	return (unsigned int *)mmPSOC_GLOBAL_CONF_PAD_SEL_53;
 }
 
+unsigned int *get_cpu_boot_status_reg(void)
+{
+	return (unsigned int *)mmPSOC_GLOBAL_CONF_WARM_REBOOT;
+}
